tests/unit/test_optimisation.py

```diff
@@ -92,6 +92,11 @@ def test_optimiser_construction(self):
             == "Covariance Matrix Adaptation Evolution Strategy (CMA-ES)"
         )
 
+        # None with no bounds
+        cost.bounds = None
+        opt = pybop.Optimisation(cost=cost)
```

Comment:
Should this check the CMAES optimiser explicitly? For GradientDescent, boundaries is None regardless of `cost.bounds` and the non-Pints optimisers do not have boundaries, is this correct?

Reply:
Yes, this will only check CMAES since it is our [default optimiser](https://github.com/pybop-team/PyBOP/blob/a66be81e409df321562c83d0df618e4821e9dbd2/pybop/optimisation.py#L51).